tests/net/socket/tcp/src/main.c

```diff
@@ -368,6 +370,95 @@ static void restore_packet_loss_ratio(void)
 		"Error setting packet drop rate");
 }
 
+void test_v4_interface_down(void)
+{
+	/* Test if the data stops transmitting after the send returned with a timeout. */
+	int c_sock;
+	int s_sock;
+	int new_sock;
+	struct sockaddr_in c_saddr;
+	struct sockaddr_in s_saddr;
+	struct sockaddr addr;
+	socklen_t addrlen = sizeof(addr);
+
+	struct timeval optval = {
+		.tv_sec = 0,
+		.tv_usec = 500000,
+	};
+
+	struct net_stats before;
+	struct net_stats after;
+	uint32_t start_time, time_diff;
+	ssize_t recved;
+	int rv;
+	uint8_t rx_buf[10];
+
+	prepare_sock_tcp_v4(CONFIG_NET_CONFIG_MY_IPV4_ADDR, ANY_PORT,
+			    &c_sock, &c_saddr);
+	prepare_sock_tcp_v4(CONFIG_NET_CONFIG_MY_IPV4_ADDR, SERVER_PORT,
+			    &s_sock, &s_saddr);
+
+	test_bind(s_sock, (struct sockaddr *)&s_saddr, sizeof(s_saddr));
+	test_listen(s_sock);
+
+	test_connect(c_sock, (struct sockaddr *)&s_saddr, sizeof(s_saddr));
+	test_send(c_sock, TEST_STR_SMALL, strlen(TEST_STR_SMALL), 0);
+
+	test_accept(s_sock, &new_sock, &addr, &addrlen);
+	zassert_equal(addrlen, sizeof(struct sockaddr_in), "wrong addrlen");
+
+	rv = setsockopt(c_sock, SOL_SOCKET, SO_RCVTIMEO, &optval,
```

Comment:
Was it really your intention to set a recv timeout on a client socket here?

Reply:
I do not have a particular preference here, mostly copied from the example which started the test, could be blocking as well

Reply:
Oops, that was supposted to be set on the new_sock instead of the c_sock, such that the test around line 427 succeeds